tests/LSP6KeyManager/tests/AllowedERC725YKeys.test.ts

```diff
@@ -1673,88 +1679,197 @@ export const shouldBehaveLikeAllowedERC725YKeys = (
-        encodeCompactBytesArray([customKey1, customKey2, zeroKey]),
-      ];
+        const permissionValues = [
+          ALL_PERMISSIONS,
+          PERMISSIONS.SETDATA,
+          encodeCompactBytesArray([customKey1, customKey2, zeroKey]),
+        ];
 
-      await setupKeyManager(context, permissionKeys, permissionValues);
-    });
+        await setupKeyManager(context, permissionKeys, permissionValues);
+      });
 
-    [
-      { allowedDataKey: customKey1 },
-      { allowedDataKey: customKey2 },
-      { allowedDataKey: zeroKey },
-    ].forEach((testCase) => {
-      it(`should pass when setting a data key listed in the allowed ERC725Y data keys: ${testCase.allowedDataKey}`, async () => {
-        const key = testCase.allowedDataKey;
-        const value = ethers.utils.hexlify(
-          ethers.utils.toUtf8Bytes("some value for " + key)
-        );
+      [{ allowedDataKey: customKey1 }, { allowedDataKey: customKey2 }].forEach(
+        (testCase) => {
+          it(`should pass when setting a data key listed in the allowed ERC725Y data keys: ${testCase.allowedDataKey}`, async () => {
+            const key = testCase.allowedDataKey;
+            const value = ethers.utils.hexlify(
+              ethers.utils.toUtf8Bytes("some value for " + key)
+            );
+
+            const setDataPayload =
+              context.universalProfile.interface.encodeFunctionData(
+                "setData(bytes32,bytes)",
+                [key, value]
+              );
+
+            await context.keyManager
+              .connect(controllerCanSetSomeKeys)
+              .execute(setDataPayload);
+
+            const result = await context.universalProfile["getData(bytes32)"](
+              key
+            );
+            expect(result).to.equal(value);
+          });
+        }
+      );
+
+      [
+        {
+          datakeyToSet: ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("Some random data key 1")
+          ),
+        },
+        {
+          datakeyToSet: ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("Some random data key 2")
+          ),
+        },
+        {
+          datakeyToSet: ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("Some random data key 3")
+          ),
+        },
+        {
+          datakeyToSet: ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("Some random data key 4")
+          ),
+        },
+        {
+          datakeyToSet: ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("Some random data key 5")
+          ),
+        },
+        {
+          datakeyToSet:
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+        },
+        {
+          datakeyToSet:
+            "0x00000000000000000000000000000000000000000000000000000000000000ca",
+        },
+      ].forEach((testCase) => {
+        it(`should revert when trying to set any random data key (e.g: ${testCase.datakeyToSet})`, async () => {
```

Comment:
Maybe we can encapsulate this in a `describe` block?

```suggestion
describe("should revert when trying to set any random data key that is not in the list of AllowedERC725YKeys")
        it(`e.g: ${testCase.datakeyToSet})`, async () => {
```